### PR TITLE
Text modeling eval

### PR DESCRIPTION
**Pull request by LeonGuertler**

### CODE REVIEW

**Review by DylanASHillier — CHANGES_REQUESTED** [review]
no killing tests tyvm

File: configs/generator/standard.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+temperature: 1.2
+top_k: 200
+max_new_tokens: 100
```
Comment (DylanASHillier) [inline configs/generator/standard.yaml:4]: [] updating this ting

File: configs/train/baseline-large.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
+model:
+  core_model_type: generic
+  num_layers: 8
```
Comment (DylanASHillier) [inline configs/train/baseline-large.yaml]: maybe change name to baseline_10 and baseline_50

File: configs/train/baseline.yaml~ [hunk 3]
```diff
@@ -0,0 +1,99 @@
+model:
+  core_model_type: generic
+  num_layers: 8
+  ffn:
+    ffn_type: generic
+    ffn_dim: 1536
```
Comment (DylanASHillier) [inline configs/train/baseline.yaml~]: rmv

File: eval.py
Comment (DylanASHillier) [inline eval.py:1]: i prefer this to be `model, _ = `

File: evals/load_evaluators.py [hunk 4]
```diff
@@ -8,11 +8,12 @@
 from evals.finetuning.qa import FinetuningQA
 from evals.mcqs.benchmarks.stories_progression import ProgressionEvaluator
 
+
```
Comment (DylanASHillier) [inline evals/load_evaluators.py]: [] (NOT DONE YET)

File: evals/mcqs/load_benchmarks.py [hunk 5]
```diff
@@ -1,24 +1,317 @@
...
+from datasets import load_dataset
+
+def get_idx_list(dataset_length, num_samples):
+    """
+    Given the dataset length and the number of samples,
+    return a list of indices to sample from the dataset
+    """
+    return np.random.choice(
+        dataset_length,
+        dataset_length if num_samples is None else min(num_samples, dataset_length),
```
Comment (DylanASHillier) [inline evals/mcqs/load_benchmarks.py]: seed pls

File: evals/mcqs/load_benchmarks.py [hunk 6]
```diff
@@ -1,24 +1,317 @@
...
+        replace=False
+    )
+
```
Comment (DylanASHillier) [inline evals/mcqs/load_benchmarks.py]: conditional happy on splitting out training eval benchmarks vs testtime eval benchmarks. "don't run popular ones during training"-Leon (so true - Dylan)

File: models/build_models.py [hunk 7]
```diff
@@ -15,8 +16,17 @@
 from models.model_heads import AutoregressiveLMHead
 from models.model_shell import ModelShell
 
+from models.experimental.weight_sharing import (
+    SharedInteriorFFNLoraAndCProj,
+    SharedInteriorFFNLora,
```
Comment (DylanASHillier) [inline models/build_models.py:20]: kill this and use my version :)

File: models/build_models.py [hunk 8]
```diff
@@ -30,18 +40,24 @@ def build_model(model_cfg=None, checkpoint=None):
...
         model.load_state_dict(checkpoint["model"])
+        current_iter = checkpoint["iter_num"]
 
     else:
         # initialize model
         model = initialize_model(model_cfg)
+        current_iter = 0
 
-    return model
+    return model, current_iter
```
Comment (DylanASHillier) [inline models/build_models.py]: training state config instead

File: models/build_models.py [hunk 9]
```diff
@@ -151,11 +188,6 @@ def initialize_model(model_cfg):
         embedding_model=embedding_model
     )
 
-    # check if embedding model weights are to be shared with the model head
-    if model_cfg["embedding_weight_tying"]:
-        # share the weights between the token embeddings and the final
-        # logit layer, following: https://paperswithcode.com/method/weight-tying
-        embedding_model.token_embedder.weight = model_head.linear.weight
```
Comment (DylanASHillier) [inline models/build_models.py:158]: d'accord

File: models/components/layers/transformer_blocks.py [hunk 10]
```diff
@@ -36,7 +36,7 @@ def __init__(self, hidden_dim, context_window, use_rope, ffn_cfg, attn_cfg):
 
         # build the ffn norm
         self.ffn_norm = build_normalization(
-            normalization_name=ffn_cfg["normalization"],
+            normalization_name=ffn_cfg.get("normalization", "rms_norm"), # Default: rms_norm
```
Comment (DylanASHillier) [inline models/components/layers/transformer_blocks.py:40]: don't like

File: models/experimental/hugging_face.py [hunk 11]
```diff
@@ -28,13 +28,13 @@ def build_model(model_cfg):
         model_str,
         trust_remote_code=True,
         attn_implementation=attn_impl,
-        torch_dtype=torch.float16,
+        torch_dtype=torch.float32,
```
Comment (DylanASHillier) [inline models/experimental/hugging_face.py]: ???? try understand me no get

File: models/experimental/hugging_face.py [hunk 12]
```diff
@@ -129,10 +129,11 @@ def __init__(self, model_cfg):
         super().__init__()
         self.model = build_model(model_cfg = model_cfg)
 
-        ## freeze the parameters
-        print("Note: Freezing the parameters of the hf_core model.")
-        for param in self.model.parameters():
-            param.requires_grad = False
+        if model_cfg.get("freeze", True):
```
Comment (DylanASHillier) [inline models/experimental/hugging_face.py]: sure

File: models/embedding_models.py [hunk 13]
```diff
@@ -115,6 +117,30 @@ def __init__(self, model_cfg):
         self.eot_token = self.tokenizer.eot_token
         self.model_cfg = model_cfg
 
+        self.dropout = torch.nn.Dropout(p=model_cfg.get("embedding_dropout", 0.0))
+
+        self.token_byte_length_cache = self._precompute_byte_lengths()
+
+    def _precompute_byte_lengths(self):
```
Comment (DylanASHillier) [inline models/embedding_models.py:125]: why was this added to the interface?

File: trainers/scheduler.py [hunk 14]
```diff
@@ -45,87 +45,3 @@ def get_lr(self, iter_num):
...
-
-class LinearDropoutScheduler(DropoutScheduler):
-    """Dropout Scheduler"""
-
-    def __init__(self, start_iter, end_iter, start_dropout_p, end_dropout_p):
-        """Initialize the dropout schedule"""
-        super().__init__(start_dropout_p)
-        self.start_iter = start_iter
-        self.end_iter = end_iter
-        self.start_dropout_p = start_dropout_p
```
Comment (DylanASHillier) [inline trainers/scheduler.py:81]: ??? why remove linear one is still useful?
